### PR TITLE
[webview_flutter] Default Android to hybrid composition

### PR DESCRIPTION
**Pull request by stuartmorgan**
Switches the default implemention on Android from virtual display to
hybrid composition. This is a breaking change.

Part of https://github.com/flutter/flutter/issues/93335

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
I've marked this as a Draft only so that we don't accidentally land it before the other things we want in 3.0 have landed; it's ready for review.

File: packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart [hunk 1]
```diff
@@ -860,192 +860,41 @@ void main() {
...
-          lastInputClientRectRelativeToViewport['right'] <=
-              viewportRectRelativeToViewport['right'],
-          isTrue);
-    }, skip: !Platform.isAndroid || _skipDueToIssue86757);
-  });
+      await loadCompleter.future;
+      final String? currentUrl = await controller.currentUrl();
+      expect(currentUrl, primaryUrl);
+    });
+  }, skip: !Platform.isAndroid || _skipDueToIssue86757);
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart:897]: Unfortunately this, like pretty much all of our other Android integration tests, currently fails :/

File: packages/webview_flutter/webview_flutter/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
-## NEXT
+## 3.0.0
 
+* **BREAKING CHANGE**: On Android, hybrid composition (SurfaceAndroidWebView)
```
Comment (blasten) [inline packages/webview_flutter/webview_flutter/CHANGELOG.md:3]: I need to think about this one, but at a high level, I don't see why changing the backend should be considered a breaking change.

Comment (stuartmorgan) [reply packages/webview_flutter/webview_flutter/CHANGELOG.md:3]: The default behavior is fundamentally changing; it's safer to do it as part of a breaking change.

**Review by blasten — APPROVED** [review]

**Comment by TheJulianJES** [comment]
Nothing major, but while this was updated: 
https://github.com/flutter/plugins/blob/6d88e3a2a1848ed93670890bc79532426b1344a2/packages/webview_flutter/webview_flutter/lib/src/webview.dart#L122

This comment wasn't updated:
https://github.com/flutter/plugins/blob/6d88e3a2a1848ed93670890bc79532426b1344a2/packages/webview_flutter/webview_flutter/lib/src/webview.dart#L115

**Pull request merged**